python/sglang/srt/mm_utils.py

```diff
@@ -164,10 +164,14 @@ def expand2square(pil_img, background_color):
     if width == height:
         return pil_img
     elif width > height:
+        if pil_img.mode == "L":
```

Comment:
Maybe just move this `if pil_img.mode` to the outer block, in case there's something else broken when `width==height`. Anyway we will need RGB.

Reply:
Makes sense! 
I moved the conversion to her as we only need to convert it if the width and height are not the same + changed `elif` to `if` condition to support the change.